api-tests/src/steps/cri-steps.ts

```diff
@@ -340,3 +422,34 @@ When(
     }
   },
 );
+
+const assertNoUnexpectedJarProperties = (
+  jarPayload: CriStubResponseJarPayload,
+  expectedNonStandardValues: string[] | undefined = undefined,
+) => {
+  for (const key of Object.keys(jarPayload)) {
+    const propertyIsExpected =
+      STANDARD_JAR_VALUES.includes(key) ||
+      expectedNonStandardValues?.includes(key);
+
+    assert.ok(
+      propertyIsExpected,
+      `Non-standard JAR property ${key} with value ${JSON.stringify(jarPayload[key as keyof typeof jarPayload])} sent to CRI but not specified in test`,
+    );
+  }
+};
+
+const STANDARD_JAR_VALUES = [
```

Comment:
(conventionally I'd put constants at the top of the module, but this is fine...)